flightros/src/racing/racing.cpp

```diff
@@ -69,8 +69,8 @@ int main(int argc, char *argv[]) {
   std::size_t num_waypoints = way_points.size();
   Eigen::VectorXd segment_times(num_waypoints);
   segment_times << 10.0, 10.0, 10.0, 10.0;
-  Eigen::VectorXd minimization_weights(num_waypoints);
-  minimization_weights << 1.0, 1.0, 1.0, 1.0;
+  Eigen::VectorXd minimization_weights(5);
```

Comment:
num_waypoints + 1 instead of constant

Reply:
I looked at the number of weights, the racing code runs with less weights than waypoints, have tried with 8 way points. The quadrotor motion is poor with two minimisation weights (0.0, 1.0). As a guess, could the weights relate to the polynomial fitting, rather than directly related to the number of segments? So keeping it at 5 means you don't have to add loads of weights if you are adding lots of waypoints...

Reply:
oh I see then I completely misinterpreted the weights. @kelia are the weights related to the polynomial fitting?

Reply:
the weights are related to the minimization order. 0th weight: position, 1st weight: velocity, ..., 4th weight: snap. 
The weights have nothing to do with the number of waypoints.